### ISSUE TITLE
Add a cleaner way to make package wrappers

### ISSUE THREAD

**Issue by ericsagnes** [NixOS/nixpkgs#13156]
The current way to make packages wrappers is to use a `*-with-plugins` package.

The original `*-with-plugin` package, [`pidgin-with-plugins`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/instant-messengers/pidgin/wrapper.nix) is mentioning that part of the code could be avoided if `buildEnv` created the directories.

It feels like the wrapper code could be factored out in a cleaner `stdenv.mkWrapper` function.

List of packages currently using this technique:

- `./pkgs/applications/audio/deadbeef/wrapper.nix`
- `./pkgs/applications/audio/puredata/wrapper.nix`
- `./pkgs/applications/networking/instant-messengers/pidgin/wrapper.nix`
- `./pkgs/tools/inputmethods/fcitx/wrapper.nix`
- `./pkgs/tools/inputmethods/ibus/wrapper.nix`
- `./pkgs/applications/misc/rxvt_unicode/wrapper.nix`

Wrappers also raise lint issues (cf #11254 ) that could be fixed by inheriting some of the base package meta attributes:

```
22:06:18.262 > ibus-with-plugins: Lacks a maintainer
22:06:18.263 > ibus-with-plugins: Lacks a platform
22:06:18.263 > ibus-with-plugins: Lacks a license
22:06:18.263 > ibus-with-plugins: Lacks a description
```

cc `pidgin-with-plugins` creator @abbradar

**Comment by abbradar** [comment]
Truth be told, I've thought so too -- but to make it properly, we first need to get https://github.com/NixOS/nixpkgs/issues/4621 addressed -- I think we have even more such wrappers now (for more examples see e.g. `games/dwarf-fortress/dwarf-therapist/wrapper.nix`).

**Comment by zimbatm** [comment]
There is also vim_configurable and similar wrappers that try to bundle a config to an executable.

**Comment by ericsagnes** [comment]
Fixed by  #15002
